src/openloco/audio/audio.cpp

```diff
@@ -777,11 +778,75 @@ namespace openloco::audio
         return false;
     }
 
-    static void sub_48A274(vehicle* v)
+    static void sub_48A274(vehicle* baseVehicle)
     {
-        registers regs;
-        regs.esi = (int32_t)v;
-        call(0x0048A274, regs);
+        // TODO: move vehicle_26 up
+        auto v = baseVehicle->as_vehicle_2or6();
+        if (v == nullptr)
+            return;
+
+        if (v->sound_id == 0xFF)
+            return;
+
+        // TODO: left or top?
+        if (v->sprite_left == (int16_t)0x8000u)
+            return;
+
+        if (config::get().var_25 <= _112C666)
```

Comment:
This seems to be used as a limit. `var_25` gets set based on the sound quality setting, but we don't have that anymore. What should we do with it? Add a hardcoded limit, or remove the limit + counter?

Reply:
I'd keep the limit for now. Is this the only function that uses the limit?

Reply:
Yes it is. It's determined by the sound quality setting, which has been removed. It's set to 4, 8 or 16 depending on the setting. `config_t.max_sound_instances` is used in a same fashion, but doesn't seem to be used anymore in our implementation.

![image](https://user-images.githubusercontent.com/737603/55950881-36551800-5c56-11e9-9f6a-d2af15528b73.png)